.kiro/specs/nicegui-web-interface/design.md

```
@@ -0,0 +1,795 @@
+```
+
+### 2. Authentication System
+
+**Authentication Middleware (`app/ui/auth/middleware.py`)**
+
+```python
+from nicegui import app, ui
+from fastapi import Request, HTTPException
+from app.core.auth import verify_session_token
+from app.ui.auth.login import login_page
+
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)